### ISSUE TITLE
FR: Add Command for Linting Files Updated Since a Specific Date

### ISSUE THREAD

**Issue by karmeye** [platers/obsidian-linter#810]
## Is Your Feature Request Related to a Problem? Please Describe.

Regarding YAML and file system modification dates that become unsynced with each other due to moving/downloading the Vault to a new location.

Running "Lint all files" / "Lint folder" will update all files because file sys mod date is later than Yaml mod date, but the file's content hasn't actually been modified.

## Describe the Solution You'd Like

Is it in theory possible to programatically set the file system's mod date from within an Obsidian plugin?

In that case it would be nice to have a function that one runs only once the file sys mod dates gets out of sync.

The function would go through all files in the vault and make sure that the file mod date is set to the current value in whatever Yaml key represents the modification date.

## Describe Alternatives You've Considered

Is there any stand-alone app? But it would have to run on all platforms.

## Additional Context

Having run the "sync mod dates" function would allow "lint all files" to just update the  Yaml mod dates on the files that were actually modified.

**Comment by pjkaufman** [comment]
This relates to #386 . There is nothing that can be done by the Linter in regards to this. However you may be able to use Custom Commands to handle this.

**Comment by pjkaufman** [comment]
But to be clear,the Linter cannot do this because Obsidian does not allow for this in their API.

**Comment by pjkaufman** [comment]
There probably is a way to do so, but those functions are not added to an API or anything. It would take some work and testing on your part to get it working.

**Comment by pjkaufman** [comment]
> Would it be possible to have a "Lint all Files Modified Today" feature in Linter?

It would be possible so long as the date modified is consisered the source of truth. It would check if the date modidied was today and if so, then lint those files.

**Comment by pjkaufman** [comment]
Does linting on file change (#799 ) and autosave (#392 ) not address this issue? 

I am assuming the latter handles your scenario.

**Comment by pjkaufman** [comment]
Looks like I linked to the wrong issue. I was referring to #183 for autosave. It would lint the file after a change is made to it (thus no need for running a lint on the current file manually).

As for linting a file when the active file changes, that makes sense. I was pretty sure it would not meet your needs, but thought I would mention it as well.

**Comment by pjkaufman** [comment]
> > I was referring to #183 for autosave. It would lint the file after a change is made to it.
> 
> How do you detect changes made to a file?

I listen to the current active file's editor content (source mode displayed content). If the rditor says something changed, then I assume something changed.

**Comment by pjkaufman** [comment]
> > I listen to the current active file's editor content (source mode displayed content). If the rditor says something changed, then I assume something changed.
> 
> Ok.
> 
> Autosave as in #183 doesn't work for me since I don't want the file to suddenly change.
> 
> What would work is if you manage a list (as the Recent Files plugin manages a list of recent files in a json) where you add each file as soon as a change occurs. This would be a list of files pending linting. Then make a command available that lints all files in the list and then clears it.
> 
> Then there's no need to iterate through all files as suggested above.
> 
> But this list of files pending linting might cause problems when multiple clones of the same vault exists, where files haven't been synced.

I can say that I would not add that data to a json file if I were to add it. So there would be no issue with syncing.

It sounds like you are looking for a feature that currently has not been requested, so I can put it in the backlog and see what interest current users have in this feature. I am open to PRs in the meantime, but I at this time I don't see myself working on this anytime soon.

**Comment by pjkaufman** [comment]
Sounds good. I am glad that you found a solution.
